File: python/clock/clock.py
class Clock:
    def __init__(self, hr, min):
        self.hr = hr
        self.min = min
        self.hr += self.min // 60
        self.min %= 60
        self.hr %= 24

    def __repr__(self):
        return f"Clock({self.hr}, {self.min})"

    def __str__(self):
        return f"{self.hr:02}:{self.min:02}"

    def __eq__(self, other):
        return other.hr == self.hr and other.min == self.min

    def __add__(self, minutes):
        return Clock(self.hr, self.min + minutes)

    def __sub__(self, minutes):
        return Clock(self.hr, self.min - minutes)

File: kotlin/clock/src/main/kotlin/Clock.kt
data class Clock(private var hours: Int, private var minutes: Int) {
    private val MINUTES_IN_HOUR = 60
    private val HOURS_IN_DAY = 24

    init {
        normalize()
    }

    private fun normalize() {
        while (minutes >= MINUTES_IN_HOUR) {
            minutes -= MINUTES_IN_HOUR
            ++hours
        }
        while (minutes < 0) {
            minutes += MINUTES_IN_HOUR
            --hours
        }
        while (hours >= HOURS_IN_DAY) {
            hours -= HOURS_IN_DAY
        }
        while (hours < 0) {
            hours += HOURS_IN_DAY
        }
    }

    fun add(m: Int) {
        minutes += m
        normalize()
    }

    fun subtract(m: Int) {
        minutes -= m
        normalize()
    }

    override fun toString() = "%02d:%02d".format(hours, minutes)
}
